### PR TITLE
Automate_Strategy_List_Analytics

### PR DESCRIPTION
**Pull request by wodsuz**
- Added heap code to track and integrate with heapanalytics.
- Split strategy list elements data and create function this way more items can be added easier on StrategyData.
- Connected Atlas mine's card with /birdgeworld-claim-and-send url .
- Made a button on top for people to see the strategy list without signing up.

### CODE REVIEW

File: src/components/Strategies/StrategyList.tsx [hunk 1]
```diff
@@ -1,59 +1,39 @@
 import { Card, Row, Col, Typography } from 'antd';
 import styled from 'styled-components';
 
+import { IStrategies } from './../../types';
+
```
Comment (mzolkiewski) [inline src/components/Strategies/StrategyList.tsx:5]: Remove this empty line. Let's adhere to the convention of 'external imports; empty line; internal imports'

File: src/components/Strategies/StrategyList.tsx [hunk 2]
```diff
@@ -86,6 +66,7 @@ const Container = styled.div`
   }
 
   .soon {
```
Comment (mzolkiewski) [inline src/components/Strategies/StrategyList.tsx:68]: Rename to more descriptive `comingSoon` + add some transition so it looks nicer when it appears, e.g. `transition: opacity 0.3s ease`

File: src/components/Strategies/StrategyList.tsx [hunk 3]
```diff
@@ -1,59 +1,39 @@
...
+
 const { Meta } = Card;
 
 const { Text } = Typography;
 
 function StrategyList() {
+  const strategyClick = (strategy: IStrategies) => {
+    var comingSoonTextId = document.getElementById(strategy.title);
+    if (strategy.displayComingSoonText) {
+      comingSoonTextId!.style.display = 'block';
```
Comment (mzolkiewski) [inline src/components/Strategies/StrategyList.tsx:16]: It's generally not a good practice to manipulate styles directly. It's better to apply classes instead. Define a new sub class, e.g. `.visible` and do `opacity: 0` by default and `opacity: 1` with the class applied

File: src/components/Strategies/StrategyList.tsx [hunk 4]
```diff
@@ -1,59 +1,39 @@
...
-          <Card className="antcard" hoverable cover={<img alt="example" src="../img/ivory-tower.jpg" />}>
-            <Meta title="Send Legions Questing" description="Bridgeworld" />
-            <Text type="danger" className="soon">
-              COMING SOON!
-            </Text>
-          </Card>
-        </Col>
+        {strategies.map((strategy) => (
+          <Col span={8} key={strategy.itemId}>
+            <div onClick={() => strategyClick(strategy)}>
```
Comment (mzolkiewski) [inline src/components/Strategies/StrategyList.tsx:27]: Looks like you had to define custom events in Heap for each type of click. Would probably be easier if you just called e.g. `heap.track('StrategyClick', { name, url })`

File: src/components/Header/HeaderMain.tsx [hunk 5]
```diff
@@ -29,6 +30,7 @@ function HeaderMain() {
       </Logos>
       <FlexSpacer />
       {isAuthenticated && <Credits />}
+      <TreasureEcosystemButton />
```
Comment (mzolkiewski) [inline src/components/Header/HeaderMain.tsx:33]: Let's hide this for now

File: .env.example [hunk 6]
```diff
@@ -1,3 +1,4 @@
 REACT_APP_API_URL=http://localhost:3001
 REACT_APP_ALLOW_SIGNUP=true
-REACT_APP_MAINTENANCE_MODE=false
\ No newline at end of file
+REACT_APP_MAINTENANCE_MODE=false
+REACT_APP_HEAP_PRODUCTION_ID=111111111
```
Comment (mzolkiewski) [inline .env.example:4]: Why `PRODUCTION`? The environment itself will determine if it's production or not

**Review by mzolkiewski — CHANGES_REQUESTED** [review]

File: .env.example (hunk 6 above)
Comment (wodsuz) [reply .env.example:4]: i thought i need to declare seperate id's for production and development

**Review by mzolkiewski — CHANGES_REQUESTED** [review]

File: src/components/Strategies/StrategyList.tsx [hunk 7]
```diff
@@ -85,8 +66,24 @@ const Container = styled.div`
     display: inline-block;
   }
 
-  .soon {
+  .comingSoon {
     float: right;
+    animation: opacity 0.5s ease;
+  }
+
+  .visible {
```
Comment (mzolkiewski) [inline src/components/Strategies/StrategyList.tsx:74]: Another thing - since `.visible` applies only to `.comingSoon` then move it inside `.comingSoon` (you can nest CSS here)

File: public/index.html [hunk 8]
```diff
@@ -36,6 +36,39 @@
...
+                'track',
+                'unsetEventProperty',
+              ],
+              o = 0;
+            o < p.length;
+            o++
+          )
+            heap[p[o]] = n(p[o]);
+        });
+      heap.load('%REACT_APP_HEAP_PROJECT_ID%');
```
Comment (mzolkiewski) [inline public/index.html:70]: This name does not match the name in `.env`

File: src/components/Strategies/StrategyList.tsx [hunk 9]
```diff
@@ -1,59 +1,45 @@
...
-            </Text>
-          </Card>
-        </Col>
+        {strategies.map((strategy) => (
+          <Col span={8} key={strategy.itemId}>
+            <div onClick={() => strategyClick(strategy)}>
+              <Card hoverable cover={<img alt={strategy.title} src={strategy.imageSrc} />}>
+                <Meta title="Claim Rewards" description={strategy.title} />
+                <Text type="danger" className="comingSoonText">
+                  <div className="invisible" id={strategy.title}>
```
Comment (mzolkiewski) [inline src/components/Strategies/StrategyList.tsx:35]: Why the nested div?

Also, since you have the `id` in `strategy` why not use that? It's guaranteed to be unique

File: src/components/Strategies/StrategyList.tsx [hunk 10]
```diff
@@ -85,8 +71,12 @@ const Container = styled.div`
     display: inline-block;
   }
 
-  .soon {
+  .comingSoonText {
```
Comment (mzolkiewski) [inline src/components/Strategies/StrategyList.tsx:74]: You still haven't set the start and end opacity. Does it actually animate like this? Whatever the case, I already explained that you need to explicitly set both start and end.

File: src/components/Strategies/strategyData.tsx [hunk 11]
```diff
@@ -0,0 +1,45 @@
+import { IStrategies } from '../../types';
+export const strategies: IStrategies[] = [
+  {
+    title: 'Bridgeworld (Atlas Mine)',
+    imageSrc: '../img/atlas-mine.jpg',
+    displayComingSoonText: false,
```
Comment (mzolkiewski) [inline src/components/Strategies/strategyData.tsx:6]: Let's just shorten this to `comingSoon`. A good descriptive name is as short as possible - coming up with a good short name is very hard and you might be tempted to useEntiteSentencesToNameStuff but the longer it is the harder it is to comprehend

File: src/components/Strategies/strategyData.tsx [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+    itemId: 1,
```
Comment (mzolkiewski) [inline src/components/Strategies/strategyData.tsx:7]: `id`

**Review by mzolkiewski — CHANGES_REQUESTED** [review]

File: src/components/Strategies/StrategyList.tsx [hunk 13]
```diff
@@ -122,8 +94,15 @@ const Container = styled.div`
...
     float: right;
+    transition: opacity 0.2s ease;
+  }
+  .comingSoonText-invisible {
+    opacity: 0;
+  }
+  .comingSoonText-visible {
```
Comment (mzolkiewski) [inline src/components/Strategies/StrategyList.tsx:104]: You are really overthinking / overengineering it. Let me spell it out for you:

```css
.comingSoonText {
  opacity: 0;
  ...

  .visible {
     opacity: 1;
  }
}
```
Since the CSS is nested, `styled-components` will automatically transform it so that the resulting CSS will be

```css
.comingSoonText {
  opacity: 0;
  ...
}
.comingSoonText.visible {
  opacity: 1;
}
```

And then you just `comingSoonText!.classList.add('visible');`

File: src/components/Strategies/StrategyList.tsx [hunk 14]
```diff
@@ -2,75 +2,47 @@ import { Card, Row, Col, Typography } from 'antd';
...
+    try {
+      (window as any).heap.track('StrategyClicked', {
+        strategy: strategy.title,
+        url: strategy.detailPageURL,
+        id: strategy.itemid,
+      });
+    } catch (e) {
+      console.error(e as any);
+    }
+    const comingSoonTextId = document.getElementById(String(strategy.itemid));
```
Comment (mzolkiewski) [inline src/components/Strategies/StrategyList.tsx:23]: Why `Id`? Id implies a number and this a HTML element. Just do `comingSoonText` or even `comingSoon`.
You really need to think about proper naming

**Review by mzolkiewski — CHANGES_REQUESTED** [review]

File: src/components/Strategies/strategyData.tsx [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+    title: 'Tales of Elleria',
+    imageSrc: './img/tales-of-alleria.jpg',
+    ComingSoon: true,
+    itemid: 4,
+    detailPageURL: '',
+  },
+  {
+    title: 'SmithyDAO',
+    imageSrc: '../img/smithy.jpg',
+    ComingSoon: true,
```
Comment (mzolkiewski) [inline src/components/Strategies/strategyData.tsx:34]: `comingSoon` - why capitalize? You need to be consistent and pay more attention to this

File: src/components/Strategies/StrategyList.tsx (hunk 13 above)
Comment (wodsuz) [reply src/components/Strategies/StrategyList.tsx:104]: thank you apologies for overthinking about this.

File: src/components/Strategies/StrategyList.tsx (hunk 14 above)
Comment (wodsuz) [reply src/components/Strategies/StrategyList.tsx:23]: i think i did this because of misunderstanding this https://github.com/chronologic/automate-ui/pull/44#discussion_r864544882 changed it back to strategy.title since unique for each card.

File: src/components/Strategies/strategyData.tsx [hunk 16]
```diff
@@ -0,0 +1,45 @@
+import { IStrategies } from '../../types';
+export const strategies: IStrategies[] = [
+  {
+    title: 'Bridgeworld (Atlas Mine)',
+    imageSrc: '../img/atlas-mine.jpg',
```
Comment (mzolkiewski) [inline src/components/Strategies/strategyData.tsx:5]: `..` here and `.` in some other items. Make it consistent
